test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -211,49 +219,45 @@ def test_emacs(image):
 @pytest.mark.model("N/A")
 def test_cuda_paths(gpu):
     """
-    Test to ensure directory structure for GPU files has cuda version in it
+    Test to ensure directory structure for GPU Dockerfiles has cuda version in it
 
     :param gpu: gpu image uris
     """
     image = gpu
     if "example" in image:
-        pytest.skip("Skipping example containers which are not explicitly tied to a cuda version")
+        pytest.skip("Skipping Example Dockerfiles which are not explicitly tied to a cuda version")
 
-    dlc_path = os.getcwd().split('/test/')[0]
+    dlc_path = os.getcwd().split("/test/")[0]
     job_type = "training" if "training" in image else "inference"
 
     # Ensure that image has a supported framework
     frameworks = ("tensorflow", "pytorch", "mxnet")
-    framework = ''
+    framework = ""
     for fw in frameworks:
         if fw in image:
             framework = fw
             break
     assert framework, f"Cannot find any frameworks {frameworks} in image uri {image}"
 
-    # Get cuda and framework version through regex
-    cuda_version_match = re.search(r'-cu\d+-', image)
-    cuda_version = cuda_version_match.group().strip('-')
-
-    framework_version_match = re.search(r':\d+.\d+.\d+', image)
-    framework_version = framework_version_match.group().strip(':')
+    # Get cuda, framework version, python version through regex
+    cuda_version = re.search(r"-(cu\d+)-", image).group(1)
+    framework_version = re.search(r":(\d+(.\d+){2})", image).group(1)
+    python_version = re.search(r"(py\d+)", image).group(1)
```

Comment:
nit: For python_version, you don't need to use groups, though there is no harm in using it.